### PR TITLE
Bug 905047 - [bb][refactor] Buttons

### PR DESCRIPTION
**Pull request by basiclines**
- Splitted `buttons.css` into 4 files `layout.css`, `icons.css`, `skin.default.css`, `skin.flat.css` (was lists buttons)
- Removed `compact` case as is no longer valid (deprecated)
- Disabled states are now achieved by `rgba` color model, in this way we avoid double declarations for "dark" and "white" backgrounds.
- Removed duplicated code for icons active states.
- `bbs-name` is now used as a namespace for skins (Building Block Skin) because of obvious possible collissions with main component namespaces. This change has been applied to `headers.css` aswell.
- Removed no longer needed images (disabled states)

### CODE REVIEW

**Comment by rik** [comment]
Instead of using an image for the shadow, could you try changing `border-bottom` to the top color of the image and using `:after` for the other color?

File: style/buttons/skin.default.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  text-shadow: none;
+  border-color: #820000;
+}
+
+.bb-button.danger:active { /* Danger Press */
+  background-image: url(images/ui/danger-press.png);
+  background-color: #890707;
+}
+
+.bb-button[disabled] {
```
Comment (rik) [inline style/buttons/skin.default.css]: Use `:disabled`.

File: style/buttons/skin.flat.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+.bbs-flat .bb-button:active { /* Press state */
+  background-color: #008AAA;
+}
+
+.bbs-flat .bb-button:active:after {
+  opacity: 0;
+}
+
+.bbs-flat .bb-button[aria-disabled="true"]:after { /* Disabled */
```
Comment (rik) [inline style/buttons/skin.flat.css]: Is this obsolete or should it change to `:disabled` ?

File: style/buttons/index.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  </div>
+
+  <h2 class="bb-docs">Button List </h2>
+  <div class="bb-docs bbs-flat">
+    <ul>
+      <li>
+        <button class="bb-button">Default</button>
+      </li>
+      <li>
+        <button class="bb-button" disabled="disabled">Disabled</button>
```
Comment (rik) [inline style/buttons/index.html]: Please remove all the `="disabled"` in this file.

**Comment by pivanov** [comment]
for the `shadow` thing we can use multiple borders

**Comment by basiclines** [comment]
@pivanov i've tried that but the `border-radius` gets choppy :S i think we must keep the shadow hack in this case

**Comment by basiclines** [comment]
@Rik The trick with border instead of image works well!

**Comment by rnowm** [comment]
Hey sorry I could not review that as I was on vacation :) but I was now checking the disabled states, and when we created the current BB the disabled danger button was reddish, is this something you reviewed with the visual design team? cc/ @sergivila

**Comment by basiclines** [comment]
Sure @sergivila is aware of this change :D

**Pull request merged**